pkg/coreos/coreos.go

```diff
@@ -131,3 +159,38 @@ func (c *coreos) FetchCoreOSStream() (map[string]any, error) {
 
 	return m, nil
 }
+
+func generateCompressedCPIO(fileContent []byte, filePath string, mode cpio.FileMode) ([]byte, error) {
```

Comment:
https://github.com/openshift/assisted-image-service/pull/457 should make it possible to avoid copying-and-pasting this code here. Then you can use [`isoeditor.NewInitRamFSStreamReaderFromISO()`](https://github.com/openshift/assisted-image-service/blob/1b29aba9d5bee83a536e40233a8170dbb2c72520/pkg/isoeditor/initramfs.go#L21) to create a stream to write over the initrd.